src/lib/comms/include/sol-lwm2m.h

```diff
@@ -733,10 +755,34 @@ void sol_lwm2m_resource_clear(struct sol_lwm2m_resource *resource);
  * before calling this function.
  */
 int sol_lwm2m_resource_init(struct sol_lwm2m_resource *resource,
-    uint16_t id, uint16_t resource_len,
+    uint16_t id, enum sol_lwm2m_resource_type type, uint16_t resource_len,
     enum sol_lwm2m_resource_data_type data_type, ...);
 
 /**
+ * @brief Initializes a LWM2M resource of type multiple using two @c sol_vector.
+ *
+ * This function makes it easier to init a LWM2M resource of type multiple,
+ * when we don't know beforehand the amount of Resource Instances desired.
+ * The last argument is a @c sol_vector holding elements of the relevant
+ * resource data type as per the table in @see sol_lwm2m_resource_init().
```

Comment:
Not very good explanation. You know how many resources instances you have, since you pass it ass vectors.